core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -72,162 +100,315 @@ sealed abstract class Parser[+A] {
       Left(Parser.Error(offset, Parser.Expectation.unify(NonEmptyList.fromListUnsafe(err.toList))))
   }
 
-  /** If this returns an epsilon failure, change to success with
-    * a value of None, else Some for the parsed value
-    */
+  /**
+   * Convert epsilon failures into None values.
+   *
+   * Normally if a parser fails to consume any input it fails with an
+   * epsilon failure. The `?` method converts these failures into
+   * None values (and wraps other values in `Some(_)`).
+   *
+   * If the underlying parser failed with other errors, this parser
+   * will still fail.
+   */
   def ? : Parser[Option[A]] =
     Parser.oneOf(Parser.map(this)(Some(_)) :: Parser.Impl.optTail)
 
-  /** Don't capture anything, just do the parsing
-    * This can be a significant optimization in that
-    * it removes internal allocations as much as possible.
-    *
-    * see also Functor.as(a) which is .void.map(_ => a)
-    * or a *> b and a <* b which use void to remove allocations
-    * for the side that is discarded.
-    */
+  /**
+   * Parse without capturing values.
+   *
+   * Calling `void` on a parser can be a significant optimization --
+   * it allows the parser to avoid allocating results to return.
+   *
+   * Other methods like `as`, `*>`, and `<*` use `void` internally to
+   * discard allocations, since they will ignore the original parsed
+   * result.
+   */
   def void: Parser[Unit] =
     Parser.void(this)
 
-  /** Discard the value and get the substring that this parser
-    * matches
-    */
+  /**
+   * Return the string matched by this parser.
+   *
+   * When parsing an input string that the underlying parser matches,
+   * this parser will return the matched substring instead of any
+   * value that the underlying parser would have returned. It will
+   * still match exactly the same inputs as the original parser.
```

Comment:
You might mention this is very efficient and the same as .void.string internally.